javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -560,7 +560,10 @@ private ReturnTypeHolder getReturnTypes(Operation operation, boolean isProtocolM
 
         IType responseBodyType = SchemaUtil.getOperationResponseType(operation, settings);
         if (isProtocolMethod) {
-            if (responseBodyType instanceof ClassType || responseBodyType instanceof ListType || responseBodyType instanceof MapType) {
+            if (responseBodyType instanceof ClassType
+                    || responseBodyType instanceof ListType
+                    || responseBodyType instanceof MapType
+                    || responseBodyType == GenericType.FluxByteBuffer) {
                 responseBodyType = ClassType.BinaryData;
```

Comment:
`SchemaUtil.getOperationResponseType` handles `BinaryResponse`.

But m4 does not use `BinaryResponse` for `text/csv`.